### PR TITLE
[14540] Reorganization of examples and dir name changes

### PR DESCRIPTION
**Pull request by antonrv**
## Description
This PR deprecates the examples of the previous PubSub interface and changes the names of the directories, also reorganizing corresponding CMake scripts.

- Dir renaming: `C++/` -> `cpp/`
- Dir renaming: `C++/DDS/` -> `cpp/dds/`
- Dir creation: `cpp/rtps/` containing previous `C++/RTPSTest_*` examples
- Dir examples renaming: `RTPSTest_*` examples without `RTPSTest_ prefix`, following the convention of *dds* examples (converting to CamelCase):
  - `C++/RTPSTest_as_socket/` -> `cpp/rtps/AsSocket/`
  - `C++/RTPSTest_persistent/` -> `cpp/rtps/Persistent/`
  - `C++/RTPSTest_registered/` -> `cpp/rtps/Registered/`
- Deletion (due to interface deprecation) of all examples using previous *PubSub* interface (corresponding examples had been already migrated to `C++/DDS/` dir with the new *Fast-DDS* interface.
- Reflect those directory changes (and target binaries and directories) in corresponding CMakeLists.txt files.
- Minor reorganization in CMakeLists.txt files to clearly encapsulate examples in `cpp/dds/CMakeLists.txt` and `cpp/rtps/CMakeLists.txt`.

After approval of this PR, then PR #369 in Fast-DDS-docs, associated to task #14543, should be reviewed to maintain consistency across Fast-DDS and Fast-DDS-docs repos.

## HelloWorldExample revamping

The DDSHelloWorldExample was doing pretty odd things. To summarise:
- On participant creation the method `DomainParticipantFactory::load_profiles()` is called always. That replaces the
  default profiles (as `default_participant_qos_`) if there is an xml profile file and the defaults are defined there.

- If on entity creation we use the `XXXXX_QOS_DEFAULT` value which is a global constant initialized with the default
  values. It ends up doing:  
```c++
    const DomainParticipantQos& pqos = (&qos == &PARTICIPANT_QOS_DEFAULT) ? default_participant_qos_ : qos;
```
 That is, if it is passed:
 + the object `XXXXX_QOS_DEFAULT` it loads the xml profile values
 + a copy of `XXXXX_QOS_DEFAULT` it ignores the xml profile values and get library defaults.

 In the master implementation both behaviors are mixed:
 - The participants and the DataReaders are using the default qos
 - the endpoints are using the profile loaded qos

The example was modified to:
 - use default library values by default
 - use xml profile values if the flag **-e/--env** is enforced
 - traditionally participant names were set and subscriber was created reliable. This behavior is only kept if profile values are not loaded. This way they can be modified by the user.
 - For backward compatibility, the original position-based cli is maintained but not documented (is deprecated).

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by MiguelBarro — CHANGES_REQUESTED** [review]
Some issues:
+ Modify the project name in files:
```
        examples\cpp\dds\SampleConfig_Multimedia\CMakeLists.txt
        examples\cpp\dds\SampleConfig_Events\CMakeLists.txt
        examples\cpp\dds\SampleConfig_Controller\CMakeLists.txt
        examples\cpp\dds\Keys\CMakeLists.txt
        examples\cpp\dds\HistoryKind\CMakeLists.txt
```
+ Some examples have been removed:
  - `examples/C++/UserDefinedTransportExample` That's ok because HelloWorldExampleSharedMem replicates behavior.
  -  `examples/C++/XMLProfiles` That's wrong because there is no DDS example that shows this feature. A workaround may be to modify the `HelloWorldExample` cli to admit profiles. This requires discussion on *The Council of Elbrus*.
+ There is a faulty example:
👍 DDSWriterLoansExample
👍 DDSZeroCopyExample
👍 DDSHelloWorldExampleDataSharing 
👍 DDSHelloWorldExampleSharedMem
👍 DDSHelloWorldExampleTCP
👍 DDSSecureHelloWorldExample
👍 DDSBenchmark
👍 DDSClientServerTest
👍 DDSDeadlineQoSExample
👍 DDSDisablePositiveACKs
👍 DDSDynamicHelloWorldExample
👍 DDSFiltering
👍 DDSFlowControlExample
👍 DDSLifespanQoSExample
👍 DDSLivelinessQoS
👍 DDSOwnershipStrengthQoSExample
❌ DDSStaticHelloWorldExample 💨 HelloWorldSubscriber.xml is not installed (`update examples/cpp/dds/StaticHelloWorldExample/CMakeLists.txt`)
👍 DDSConfigurability (former DDSUseCase)
👍 DDSHistoryKind
👍 DDSKeys
👍 DDSLateJoiners
👍 DDSSampleConfig_Controller
👍 DDSSampleConfig_Events
👍 DDSSampleConfig_Multimedia
👍 DDSCustomListenerExample
👍 TypeLookupExample
👍 TypeLookupExample

**Comment by MiguelCompany** [comment]
The issue with the XML on the StaticHelloWorld is already reported on internal ticket 13328

**Comment by MiguelBarro** [comment]
*The Council of Elbrus* decided the `examples/C++/XMLProfiles` will be removed and the HelloWorldExample cli would be extended with an option that allows loading *xml* profiles.

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Review by MiguelBarro — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

File: examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp [hunk 1]
```diff
@@ -21,42 +21,202 @@
...
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a numeric argument\n");
```
Comment (MiguelCompany) [inline examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp:101]: requires a**n** ~~numeric~~ argument

File: examples/cpp/dds/HelloWorldExample/README.txt [hunk 2]
```diff
@@ -3,3 +3,18 @@ To launch this test open two different consoles:
 In the first one launch: ./DDSHelloWorldExample publisher (or DDSHelloWorldExample.exe publisher on windows).
 In the second one: ./DDSHelloWorldExample subscriber (or DDSHelloWorldExample.exe subscriber on windows).
 
+In order to use xml profiles (--env or shorcut -e cli flags):
+    - reference the xml profiles file setting the environment variable FASTRTPS_DEFAULT_PROFILES_FILE to its path.
+    - called it DEFAULT_FASTRTPS_PROFILES.xml and make sure the file is besides the DDSHelloWorldExample binary.
```
Comment (MiguelCompany) [inline examples/cpp/dds/HelloWorldExample/README.txt:8]: name it

File: examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp [hunk 3]
```diff
@@ -156,68 +156,88 @@ int main(
...
-        if (parse.error())
+        else
         {
+            option::printUsage(fwrite, stdout, usage, columns);
             return 1;
         }
+    }
 
-        if (options[HELP])
+    // Decide among the old and new syntax
```
Comment (MiguelCompany) [inline examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp:203]: ~~among~~ -> between

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Three nits

File: examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp (hunk 1 above)
Comment (MiguelBarro) [reply examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp:101]: ack

File: examples/cpp/dds/HelloWorldExample/README.txt (hunk 2 above)
Comment (MiguelBarro) [reply examples/cpp/dds/HelloWorldExample/README.txt:8]: ack

File: examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp (hunk 3 above)
Comment (MiguelBarro) [reply examples/cpp/dds/HelloWorldExample/HelloWorld_main.cpp:203]: ack

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Pull request merged**